packages/core/contracts/financial-templates/common/financial-product-libraries/contract-for-difference-libraries/RangeBondContractForDifferenceFinancialProductLibrary.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/math/SignedSafeMath.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import "./ContractForDifferenceFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+interface ContractForDifferenceInterface {
+    function collateralPerPair() external view returns (uint256);
+}
+
+contract RangeBondContractForDifferenceFinancialProductLibrary is
+    ContractForDifferenceFinancialProductLibrary,
+    Lockable
+{
+    using FixedPoint for FixedPoint.Unsigned;
+    using SignedSafeMath for int256;
+
+    struct RangeBondContractForDifferenceParameters {
+        uint256 bondNotional;
+        uint256 highPriceRange;
+        uint256 lowPriceRange;
+    }
+
+    mapping(address => RangeBondContractForDifferenceParameters) public contractForDifferenceParameters;
+
+    function setContractForDifferenceParameters(
+        address contractForDifferenceAddress,
+        uint256 bondNotional,
+        uint256 highPriceRange,
+        uint256 lowPriceRange
+    ) public nonReentrant() {
+        require(
+            ExpiringContractInterface(contractForDifferenceAddress).expirationTimestamp() != 0,
+            "Invalid CFD address"
+        );
+        require(highPriceRange > lowPriceRange, "Invalid bounds");
+
+        RangeBondContractForDifferenceParameters memory params =
+            contractForDifferenceParameters[contractForDifferenceAddress];
+        require(
+            params.bondNotional == 0 && params.lowPriceRange == 0 && params.highPriceRange == 0,
+            "Parameters already set"
+        );
+
+        contractForDifferenceParameters[contractForDifferenceAddress] = RangeBondContractForDifferenceParameters({
+            bondNotional: bondNotional,
```

Comment:
Does this mean we're ok if the bondNotional is set to 0?

Reply:
previously this would be possible. I've added some additional checks that would make this possible.